src/sources/models.js

```diff
@@ -139,10 +141,12 @@ class Source {
 }
 
 class Resource {
-  constructor(source, pathParts) {
+  constructor(source, pathParts, versionDescriptor=LATEST) {
```

Comment:
Is `LATEST` encoding anything other than the value not being specified? If not, I suggest defaulting to `versionDescriptor=undefined`. That would simplify some of the comparisons, such as:

```diff
- if (this.versionDescriptor && this.versionDescriptor!==LATEST) {
+ if (this.versionDescriptor) {

- if (versionDescriptor===LATEST)
+ if (!versionDescriptor)
```

Reply:
> Is LATEST encoding anything other than the value not being specified?

No, and `LATEST` is the default value as well. The verbose checks are just defensive programming. 

I can change this to `undefined` if you think it would improve the maintainability of the code? I personally found it useful to read the code as LATEST not undefined, but I'm happy to change if it's simpler for others.

Reply:
Makes sense. I'm on board with keeping `LATEST`.